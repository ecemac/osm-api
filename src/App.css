#map {
  height: 180px;
}

.leaflet-container {
  width: 100%;
  height: 100vh;
}

.search-container {
  border-radius: 5px;
  border: 0.5px solid rgb(209, 209, 209);
  width: 30%;
  overflow: scroll;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 40%;
  background-color: #fff;
  z-index: 1000;
  padding: 0 20px;
}

.wrapper {
  position: relative;
}

.search-coordinates {
  width: 30%;
  padding: 20px 0;
  border-bottom: 0.5px solid rgb(233, 233, 233);
  display: flex;
  justify-content: space-around;
  position: fixed;
  background-color: #fff;
}

.search-coordinates input {
  margin-left: 5px;
}

.search-coordinates p {
  font-weight: bold;
}

.osm-infos {
  margin-top: 70px;
}
.info {
  padding-top: 10px;
  border-bottom: 0.5px solid rgb(209, 209, 209);
}

.info button {
  text-align: left;
  color: rgb(26, 26, 26);
  background: none;
  border: none;
  font-size: 14px;
  padding: 0;
}

.info p {
  font-size: 12px;
  color: rgb(107, 107, 107);
}
