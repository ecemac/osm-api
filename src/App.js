import React, { useEffect, useState } from "react";
import "./App.css";
import osmtogeojson from "osmtogeojson";
import DeckGL from "@deck.gl/react";
import { EditableGeoJsonLayer, DrawPolygonMode } from "nebula.gl";
import { StaticMap } from "react-map-gl";
import ReactMapboxGl, { Layer, Feature } from "react-mapbox-gl";
import "mapbox-gl/dist/mapbox-gl.css";

const MAPBOX_ACCESS_TOKEN =
  "pk.eyJ1IjoiZWNlbWFjIiwiYSI6ImNreGF5djhzbzA5ZHkycnBkODByZHZob2MifQ.qqHky7ydWDMi0C48FTHX8A";

const initialViewState = {
  longitude: -122.43,
  latitude: 37.775,
  zoom: 12,
};

function App() {
  const [featureCollection, setFeatureCollection] = useState({
    type: "FeatureCollection",
    features: [],
  });
  const [selectedFeatureIndexes] = useState([]);

  /* useEffect(() => {
    fetch("https://www.openstreetmap.org/api/0.6/map?bbox=-0.4,51,0.2,51", {
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
    })
      .then((response) => response.json())
      .then((data) => {
        let osm = osmtogeojson(data);
        setFeatureCollection(osm);
      });
  }, []); */

  console.log("OSM: ", featureCollection);

  const layer = new EditableGeoJsonLayer({
    data: featureCollection,
    mode: DrawPolygonMode,
    selectedFeatureIndexes,
    onEdit: ({ updatedData }) => {
      setFeatureCollection(updatedData);
    },
  });

  console.log("selected index: ", selectedFeatureIndexes);

  return (
    <div className="App">
      <DeckGL
        initialViewState={initialViewState}
        controller={{
          doubleClickZoom: false,
        }}
        layers={[layer]}
        getCursor={layer.getCursor.bind(layer)}
      >
        <StaticMap mapboxApiAccessToken={MAPBOX_ACCESS_TOKEN} />
      </DeckGL>
    </div>
  );
}

export default App;
