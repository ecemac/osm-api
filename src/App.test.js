import React from "react";
import ReactDOM from "react-dom";
import { screen, render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import App, { findBbox } from "./App";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

it("matched snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders osm data", async () => {
  const fakeOsmData = {
    type: "FeatureCollection",
    features: [],
  };
  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeOsmData),
    })
  );

  global.fetch.mockRestore();
});

it("should render loading", () => {
  render(<App />);
  const loadingElement = screen.getByTestId("loading");
  expect(loadingElement).toBeInTheDocument();
  expect(loadingElement).toHaveTextContent("Loading...");
});

it("should return bbox", () => {
  const value = findBbox(0, 0);
  expect(value).toStrictEqual([-0.0025, -0.0025, 0.0025, 0.0025]);
});
