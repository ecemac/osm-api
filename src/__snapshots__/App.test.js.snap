// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div
  id="map"
>
  <div />
  <div
    className="search-container"
  >
    <form
      className="search-coordinates"
      onSubmit={[Function]}
    >
      <label>
        Lat:
        <input
          defaultValue={52.520008}
          name="lat"
          onChange={[Function]}
          type="text"
        />
      </label>
      <label>
        Long:
        <input
          defaultValue={13.404954}
          name="long"
          onChange={[Function]}
          type="text"
        />
      </label>
      <input
        type="submit"
        value="Go"
      />
    </form>
    <div
      className="osm-infos"
    >
      <p
        data-testid="loading"
      >
        Loading...
      </p>
    </div>
  </div>
</div>
`;
